docker/install-dynamic-plugins.py

```diff
@@ -181,8 +185,39 @@ def verify_package_integrity(plugin: dict, archive: str, working_directory: str)
     if hash_digest != output.decode('utf-8').strip():
       raise InstallException(f'{package}: The hash of the downloaded package {output.decode("utf-8").strip()} does not match the provided integrity hash {hash_digest} provided in the configuration file')
 
+# Create the lock file, so that other instances of the script will wait for this one to finish
+def create_lock(lock_file_path):
+    while True:
+      try:
+        with open(lock_file_path, 'x'):
+          print(f"======= Created lock file: {lock_file_path}")
+          return
+      except FileExistsError:
+        wait_for_lock_release(lock_file_path)
+
+# Remove the lock file
+def remove_lock(lock_file_path):
+   os.remove(lock_file_path)
+   print(f"======= Removed lock file: {lock_file_path}")
+
+# Wait for the lock file to be released
+def wait_for_lock_release(lock_file_path):
+   print("======= Waiting for lock release...")
```

Comment:
```suggestion
   print("======= Waiting for lock release...", flush=True)
```

I tested this with 2 replicas, and noticed that one of the pods was waiting (expected) but with nothing displayed in the init container logs. So I was wondering if the container was stuck. I checked that the pod was scheduled, and then when it acquired the lock, all the logs would show up.
I guess if we had [flushed the output](https://docs.python.org/3/library/functions.html#print), I would have seen this immediately in the logs, giving immediate feedback to the users.

Reply:
Works better with the latest changes 👍🏿. We now have immediate feedback in the logs of the pods waiting for the lock.